lib/StripeResource.js

```diff
@@ -381,13 +381,16 @@ StripeResource.prototype = {
         }
       });
 
-      req.on('socket', (socket) => {
+      req.once('socket', (socket) => {
```

Comment:
I'm not 100% sure that `'socket'` event can only be sent once, and the body of the function seems to imply it might be more than once. Do you know of a guarantee that this is at-most-once?

Reply:
Me neither :-) You never can be 100% sure in something in software. But look at the code - it checks a boolean flag `socket.connecting` (https://nodejs.org/api/net.html#net_socket_connecting) and in case of `true` waits for connection to be established by setting other event listener (`socket.once`). 
In both cases this function are finishing with `res.end()` - so, I'm pretty sure it's designed to be running by `once`.
 I can keep an `on`, as you wish, but see no error in this refactoring.

Reply:
In fact, not, keeping `on` is an error. Just trace what will happen in this event handler will be called more than once - it will attempt to write to the `req` after calling `req.end` in previous call or in `socket.once` event handler.

Reply:
Okay great, I think that looks right. Thanks for digging here.